components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1430,12 +1432,50 @@ void RewardsServiceImpl::SetCatalogIssuers(const std::string& json) {
   bat_ledger_->SetCatalogIssuers(json);
 }
 
+std::vector<uint64_t> RewardsServiceImpl::GetEarningsRange() {
+  auto one_day_in_seconds = base::Time::kMicrosecondsPerDay /
+      base::Time::kMicrosecondsPerSecond;
+
+  auto now = base::Time::Now();
+  base::Time::Exploded exploded;
+  now.LocalExplode(&exploded);
+
+  uint64_t offset_in_seconds;
+  if (exploded.day_of_month > 5) {
+    offset_in_seconds = (exploded.day_of_month - 5) * one_day_in_seconds;
+  } else {
+    offset_in_seconds = exploded.day_of_month * one_day_in_seconds;
+  }
+
+  auto now_in_seconds = (now - base::Time()).InSeconds();
+  uint64_t from_timestamp = now_in_seconds - offset_in_seconds;
+  uint64_t to_timestamp = now_in_seconds;
+
+  return std::vector<uint64_t>({
```

Comment:
I would use std::pair here if the vector always contains two elements

Reply:
👍 fixed